src/Rules/NameUniqueForAirlines.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Rules;
+
+use Closure;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Lightit\Backoffice\Airline\Domain\Models\Airline;
+
+class NameUniqueForAirlines implements ValidationRule
+{
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+        /** @phpstan-ignore-next-line */
+        $airline = request()->route('airline');
+        /** @var int $airlineId */
+        $airlineId = $airline->id ?? null;
+        $existingAirline = Airline::where('name', $value)->first();
+
+        if ($airlineId) {
+            $currentAirline = Airline::query()->find($airlineId);
+            if ($currentAirline && $currentAirline->name === $value) {
+                return;
+            }
+        }
+        if ($existingAirline && $existingAirline->id != $airlineId) {
+            $fail('The airline name must be unique');
+        }
+    }
```

Comment:
I feel like there is an easier way to do this
get the airline from the route
search for the airline with the $value
if the ids are the same, return
if the ids are different, fail

no need for the if current airline && name = $value